mne/preprocessing/ica.py

```diff
@@ -2171,10 +2181,13 @@ def _detect_artifacts(ica, raw, start_find, stop_find, ecg_ch, ecg_score_func,
         if isinstance(node.criterion, float):
             found = list(np.where(np.abs(scores) > node.criterion)[0])
         else:
-            found = list(np.atleast_1d(abs(scores).argsort()[node.criterion]))
+            found = list(np.atleast_1d(
+                    (-abs(scores)).argsort()[node.criterion]))
```

Comment:
use `abs(scores)[::-1].argsort()` I think

Reply:
(It ate my reply):
You mean `abs(scores).argsort()[::-1]`?
That would also work and theoretically produce less overhead, but is also less accurate with the order of NANs and same values (probably both not relevant here). See here: https://stackoverflow.com/questions/16486252/is-it-possible-to-use-argsort-in-descending-order

Reply:
exclude is an iterable of ints as of by this PR, so no NaNs.

The stackoverflow discussion also suggests [::-1] is better right?

Reply:
Well, it's sorting the scores. Can we always be sure no NaNs will occur there? Probably yes, but why bother?
The accepted answer says: "if a stable sort implementation is requested to argsort, [...] then the first strategy [== (-abs(...))] will preserve the sorting stability, but the second strategy will break stability (i.e. the positions of equal items will get reversed)."
Do you still want me to change it?

Reply:
It's just a stylistic comment.

Can we be sure no NaNs will occur? No. But we shouldn't guarantee sensible sorting when the docstring says ints and people supply something not an int either.

Reply:
Maybe I misunderstand something, but I thought the comment below the answer would imply that if any ICA-component yields a score of NaN by the method called in ica.score_sources, that component would be treated as the one with the highest score in the [::-1] version. Since we cannot be sure about the scoring func (it can also be a user provided one), I feel it's "safer" as it is.

Reply:
No, it's my fault - I had somehow thought we're dealing with picks here, not scores. You're right.